doc/tutorials/error_analysis/error_analysis_part2.ipynb

```diff
@@ -0,0 +1,550 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Numpy solution\n",
+    "time_series_1_centered = time_series_1 - np.average(time_series_1)\n",
+    "autocov = np.empty(1000)\n",
+    "\n",
+    "for j in range(1000):\n",
+    "    autocov[j] = np.dot(time_series_1_centered[:N_SAMPLES-j], time_series_1_centered[j:])\n",
+    "autocov /= N_SAMPLES\n",
+    "\n",
+    "fig = plt.figure(figsize=(10, 6))\n",
+    "plt.gca().axhline(0, color=\"gray\",linewidth=1)\n",
+    "plt.plot(autocov)\n",
+    "plt.xlabel(\"lag time $j$\")\n",
+    "plt.ylabel(\"$\\hat{R}^{XX}_j$\")\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can see that the auto-covariance function starts at a high value and decreases quickly into a long noisy tail which fluctuates around zero. The high values at short lag times indicate that there are strong correlations at short time scales, as expected. However, even though the tail looks uninteresting, it can bear important information about the statistics of your data. Small systematic deviations from 0 in the tail can be a hint that long-term correlations exist in your system. On the other hand, if there is no sign of a systematic deviation from 0 in the tail, this usually means that the correlation is decaying well within the simulation time, and that the statistics are good enough to estimate an error. In the above example, the correlation quickly decays to zero. Despite the noise in the tail, the statistics seem very reasonable."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Autocorrelation time\n",
+    "\n",
+    "Continuing our example, we can zoom into the first part of the auto-covariance function (using a log scale). We see that it indeed does have similarities with an exponential decay curve. In general, it isn't an exponential, but often can be approximated using one. If it matches reasonably well, the inverted prefactor in the exponential can be directly used as the *correlation time*, which is a measure for how many sampling intervals it takes for correlations to decay. Execute the following code cell for an illustration."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "from scipy.optimize import curve_fit\n",
+    "\n",
+    "def exp_fnc(x,a,b):\n",
+    "    return a*np.exp(-x/b)\n",
+    "\n",
+    "N_MAX = 1000\n",
+    "j = np.linspace(0,N_MAX-1,N_MAX)\n",
+    "popt, pcov = curve_fit(exp_fnc, j, autocov[:N_MAX], p0=[15,10])\n",
+    "\n",
+    "# compute analytical ACF of AR(1) process\n",
+    "AN_SIGMA_1 = np.sqrt(EPS_1 ** 2 / (1 - PHI_1 ** 2))\n",
+    "AN_TAU_EXP_1 = -1 / np.log(PHI_1)\n",
+    "an_acf_1 = AN_SIGMA_1**2*np.exp(-j/AN_TAU_EXP_1)\n",
+    "\n",
+    "fig = plt.figure(figsize=(10, 6))\n",
+    "plt.plot(j,autocov[:N_MAX], 'x', label=\"numerical ACF\")\n",
```

Comment:
Here and elsewhere, it looks like we use log scales for value ranges that include 0, which make the curves look strange. I've tried to address some of them in e54637ed47c, but you should probably have a closer look.